packages/wallets/src/features/cfd/modals/DxtradeEnterPasswordModal/DxtradeEnterPasswordModal.tsx

```diff
@@ -49,59 +69,26 @@ const DxtradeEnterPasswordModal = () => {
                 text='Create Deriv MT5 password'
             />
         );
-    };
+    }, [hide, history, isLoading, isSuccess, onSubmit, password]);
 
-    if (isMobile) {
-        return (
-            <ModalStepWrapper renderFooter={renderFooter} title={' '}>
-                {isSuccess && dxtradeAccountListSuccess && (
-                    <CFDSuccess
-                        description={successDescription}
-                        displayBalance={dxtradeBalance || ''}
-                        marketType='all'
-                        platform='dxtrade'
-                        title={`Your Deriv X${accountType === 'demo' ? ` ${accountType}` : ''} account is ready`}
-                    />
-                )}
-                {!isSuccess && (
-                    <CreatePassword
-                        icon={<DxTradePasswordIcon />}
-                        isLoading={isLoading}
-                        onPasswordChange={e => setPassword(e.target.value)}
-                        onPrimaryClick={onSubmit}
-                        password={password}
-                        platform='dxtrade'
-                    />
-                )}
-            </ModalStepWrapper>
-        );
-    }
-
-    return (
-        <ModalWrapper hideCloseButton={isSuccess}>
-            {isSuccess && dxtradeAccountListSuccess && (
+    const successComponent = useMemo(() => {
+        if (isSuccess && dxtradeAccountListSuccess) {
+            return (
                 <CFDSuccess
                     description={successDescription}
                     displayBalance={dxtradeBalance || ''}
                     marketType='all'
                     platform='dxtrade'
-                    renderButton={() => (
-                        <div className='wallets-dxtrade-enter-password__button'>
-                            <WalletButton onClick={hide} size='lg' text='Maybe later' variant='outlined' />
-                            <WalletButton
-                                onClick={() => {
-                                    hide();
-                                    history.push('/wallets/cashier/transfer');
-                                }}
-                                size='lg'
-                                text='Transfer funds'
-                            />
-                        </div>
-                    )}
+                    renderButton={() => renderFooter}
```

Comment:
```suggestion
                    renderButton={renderFooter}
```

Reply:
Its giving me type error. Because renderButton is not a react Element